src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -14,11 +14,47 @@ namespace Coverlet.Core.Helpers
 {
     internal static class InstrumentationHelper
     {
-        public static string[] GetCoverableModules(string module)
+        public static string[] GetCoverableModules(string module, string[] includeDirectories)
         {
-            IEnumerable<string> modules = Directory.EnumerateFiles(Path.GetDirectoryName(module)).Where(f => f.EndsWith(".exe") || f.EndsWith(".dll"));
-            modules = modules.Where(m => IsAssembly(m) && Path.GetFileName(m) != Path.GetFileName(module));
-            return modules.ToArray();
+            Debug.Assert(includeDirectories != null, "Parameter " + nameof(includeDirectories) + " in method " + 
+                nameof(InstrumentationHelper) + "." + nameof(GetCoverableModules) + " must not be null");
+
+            string moduleDirectory = Path.GetDirectoryName(module);
+            if (moduleDirectory == string.Empty)
+            {
+                moduleDirectory = Directory.GetCurrentDirectory();
+            }
+
+            var dirs = new List<string>(1 + includeDirectories.Length)
+            {
+                // Add the test assembly's directory.
+                moduleDirectory
+            };
+
+            // Prepare all the directories in which we probe for modules.
+            foreach (var includeDirectory in includeDirectories.Where(d => d != null))
+            {
+                var fullPath = (!Path.IsPathRooted(includeDirectory)
+                    ? Path.GetFullPath(Path.Combine(moduleDirectory, includeDirectory))
+                    : includeDirectory).TrimEnd('*');
+
+                if (!Directory.Exists(fullPath)) continue;
+
+                if (includeDirectory.EndsWith("*", StringComparison.Ordinal))
+                    dirs.AddRange(Directory.GetDirectories(fullPath));
+                else
+                    dirs.Add(fullPath);
+            }
+
+            // The test module's name must be unique.
+            var uniqueModules = new HashSet<string>
```

Comment:
This has the added benefit of preventing the test module from being included in instrumentation 👍